File: examples/elixir/examples/getting_started.ex
# List all buckets
all_buckets = ExAws.S3.list_buckets() |> ExAws.request!()

IO.puts("Buckets:")

for bucket <- all_buckets.body.buckets do
  IO.puts(bucket.name)
end

# List all objects in bucket tigris-example
bucket_name = "tigris-example"
all_objects = ExAws.S3.list_objects(bucket_name) |> ExAws.request!() |> get_in([:body, :contents])

IO.puts("Objects:")

for object <- all_objects do
  IO.puts(object.key)
end

# Put bar.txt
key = "bar.txt"
body = "Hello, world!"

response = ExAws.S3.put_object(bucket_name, key, body) |> ExAws.request!()
assert response.status_code == 200

# Read back bar.txt
response = ExAws.S3.get_object(bucket_name, key) |> ExAws.request!()
assert response.status_code == 200
assert response.body == body

File: examples/ruby/getting_started.rb
require "aws-sdk-s3"

bucket_name = "tigris-example"

s3 = Aws::S3::Client.new(
    region: "auto",
    endpoint: "https://t3.storageapi.dev",
)

# Lists all of your buckets
resp = s3.list_buckets
puts "My buckets now are:\n\n"

resp.buckets.each do |bucket|
    puts bucket.name
end

# List the first ten objects in the bucket
resp = s3.list_objects(bucket: bucket_name, max_keys: 10)
resp.contents.each do |object|
    puts "#{object.key} => #{object.etag}"
end

# Put an object into the bucket
file_name = "bar-file-#{Time.now.to_i}"
begin
    s3.put_object(
        bucket: bucket_name,
        key: file_name,
        body: File.read("getting_started.rb")
    )
    puts "Uploaded #{file_name} to #{bucket_name}."
rescue Exception => e
    puts "Failed to upload #{file_name} with error: #{e.message}"
    exit "Please fix error with file upload before continuing."
end